src/core/pdf_manager.js

```diff
@@ -14,216 +14,203 @@
-    },
+  requestRange(begin, end) {
+    unreachable('Abstract method `requestRange` called');
+  }
 
-    requestLoadedStream: function BasePdfManager_requestLoadedStream() {
-      unreachable('Abstract method `requestLoadedStream` called');
-    },
+  requestLoadedStream() {
+    unreachable('Abstract method `requestLoadedStream` called');
+  }
 
-    sendProgressiveData: function BasePdfManager_sendProgressiveData(chunk) {
-      unreachable('Abstract method `sendProgressiveData` called');
-    },
+  sendProgressiveData(chunk) {
+    unreachable('Abstract method `sendProgressiveData` called');
+  }
 
-    updatePassword: function BasePdfManager_updatePassword(password) {
-      this._password = password;
-    },
+  updatePassword(password) {
+    this._password = password;
+  }
 
-    terminate: function BasePdfManager_terminate() {
-      unreachable('Abstract method `terminate` called');
-    },
-  };
+  terminate() {
+    unreachable('Abstract method `terminate` called');
+  }
+}
 
-  return BasePdfManager;
-})();
+class LocalPdfManager extends BasePdfManager {
+  constructor(docId, data, password, evaluatorOptions, docBaseUrl) {
+    super();
 
-var LocalPdfManager = (function LocalPdfManagerClosure() {
-  function LocalPdfManager(docId, data, password, evaluatorOptions,
-                           docBaseUrl) {
     this._docId = docId;
     this._password = password;
     this._docBaseUrl = docBaseUrl;
     this.evaluatorOptions = evaluatorOptions;
-    var stream = new Stream(data);
+
+    const stream = new Stream(data);
     this.pdfDocument = new PDFDocument(this, stream);
-    this._loadedStreamCapability = createPromiseCapability();
-    this._loadedStreamCapability.resolve(stream);
+    this._loadedStreamPromise = Promise.resolve(stream);
   }
 
-  Util.inherit(LocalPdfManager, BasePdfManager, {
-    ensure: function LocalPdfManager_ensure(obj, prop, args) {
-      return new Promise(function (resolve, reject) {
-        try {
-          var value = obj[prop];
-          var result;
-          if (typeof value === 'function') {
-            result = value.apply(obj, args);
-          } else {
-            result = value;
-          }
-          resolve(result);
-        } catch (e) {
-          reject(e);
+  ensure(obj, prop, args) {
+    return new Promise(function(resolve, reject) {
+      try {
+        const value = obj[prop];
+        let result;
+        if (typeof value === 'function') {
+          result = value.apply(obj, args);
+        } else {
+          result = value;
         }
-      });
-    },
+        resolve(result);
+      } catch (e) {
+        reject(e);
+      }
+    });
+  }
 
-    requestRange: function LocalPdfManager_requestRange(begin, end) {
-      return Promise.resolve();
-    },
+  requestRange(begin, end) {
+    return Promise.resolve();
+  }
 
-    requestLoadedStream: function LocalPdfManager_requestLoadedStream() {
-      return;
-    },
+  requestLoadedStream() {
+    return;
```

Comment:
This was already in the old code; can we remove the unnecessary `return` statement (here and on the following lines too)?

Reply:
Yes, I'll take care of that.